File: apps/desktop/lib/screens/server_welcome_screen.dart
import 'package:easy_localization/easy_localization.dart';
import 'package:flutter/material.dart';
import '../design/nano_tokens.dart';
import '../widgets/nano/nano_card.dart';
import 'add_server_page.dart';

/// First-run screen: brand intro + feature highlights + "add server" CTA.
class ServerWelcomeScreen extends StatelessWidget {
  const ServerWelcomeScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final t = context.nano;
    final features = <List<dynamic>>[
      [
        Icons.dashboard_rounded,
        'welcome.featureMetricsTitle'.tr(),
        'welcome.featureMetricsDesc'.tr()
      ],
      [
        Icons.terminal_rounded,
        'welcome.featureTerminalTitle'.tr(),
        'welcome.featureTerminalDesc'.tr()
      ],
      [
        Icons.notifications_active_rounded,
        'welcome.featureAuditTitle'.tr(),
        'welcome.featureAuditDesc'.tr()
      ],
    ];
    return Scaffold(
      backgroundColor: t.bg,
      body: SafeArea(
        child: Center(
          child: ConstrainedBox(
            constraints: const BoxConstraints(maxWidth: 460),
            child: Column(
              children: [
                Expanded(
                  child: SingleChildScrollView(
                    padding: const EdgeInsets.fromLTRB(24, 48, 24, 8),
                    child: Column(
                      crossAxisAlignment: t.isIOS
                          ? CrossAxisAlignment.center
                          : CrossAxisAlignment.start,
                      children: [
                        _BrandMark(size: t.isIOS ? 72 : 64, t: t),
                        const SizedBox(height: 26),
                        Text('NanoLink',
                            textAlign:
                                t.isIOS ? TextAlign.center : TextAlign.start,
                            style: TextStyle(
                                fontSize: t.isIOS ? 32 : 36,
                                fontWeight: t.displayWeight,
                                letterSpacing: t.displayTracking,
                                height: 1.1,
                                color: t.fg)),
                        const SizedBox(height: 10),
                        Text('welcome.tagline'.tr(),
                            textAlign:
                                t.isIOS ? TextAlign.center : TextAlign.start,
                            style: TextStyle(
                                fontSize: 15.5, color: t.fg3, height: 1.5)),
                        const SizedBox(height: 28),
                        for (final f in features)
                          Padding(
                            padding: const EdgeInsets.only(bottom: 10),
                            child: NanoCard(
                              outlined: true,
                              padding: const EdgeInsets.all(14),
                              child: Row(
                                children: [
                                  NanoIconBox(f[0] as IconData,
                                      size: 40, iconSize: 20, fg: t.accent),
                                  const SizedBox(width: 14),
                                  Expanded(
                                    child: Column(
                                      crossAxisAlignment:
                                          CrossAxisAlignment.start,
                                      children: [
                                        Text(f[1] as String,
                                            style: TextStyle(
                                                fontSize: 15,
                                                fontWeight: FontWeight.w500,
                                                color: t.fg)),
                                        const SizedBox(height: 2),
                                        Text(f[2] as String,
                                            style: TextStyle(
                                                fontSize: 13, color: t.fg3)),
                                      ],
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ),
                      ],
                    ),
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.fromLTRB(24, 8, 24, 20),
                  child: Column(
                    children: [
                      SizedBox(
                        width: double.infinity,
                        height: 54,
                        child: ElevatedButton.icon(
                          onPressed: () => Navigator.of(context).push(
                            MaterialPageRoute(
                                builder: (_) => const AddServerPage()),
                          ),
                          icon: const Icon(Icons.add_rounded, size: 22),
                          label: Text('welcome.addServer'.tr(),
                              style: const TextStyle(
                                  fontSize: 16, fontWeight: FontWeight.w600)),
                          style: ElevatedButton.styleFrom(
                            backgroundColor: t.accent,
                            foregroundColor: t.onAccent,
                            elevation: 0,
                            shape: RoundedRectangleBorder(
                              borderRadius:
                                  BorderRadius.circular(t.buttonRadius),
                            ),
                          ),
                        ),
                      ),
                      if (t.isIOS)
                        Padding(
                          padding: const EdgeInsets.only(top: 14),
                          child: Text('welcome.chooseMethodHint'.tr(),
                              textAlign: TextAlign.center,
                              style: TextStyle(fontSize: 12, color: t.fg4)),
                        ),
                    ],
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}

/// Brand mark: gradient rounded square containing a 4-rounded-square grid
/// glyph with a small accent dot, matching MDBrand / IOSBrand in the mockups.
class _BrandMark extends StatelessWidget {
  const _BrandMark({required this.size, required this.t});

  final double size;
  final NanoTokens t;

  @override
  Widget build(BuildContext context) {
    return Container(
      width: size,
      height: size,
      decoration: BoxDecoration(
        gradient: LinearGradient(
          colors: [t.accent, t.tertiary],
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
        ),
        borderRadius: BorderRadius.circular(size * (t.isIOS ? 0.22 : 0.32)),
        boxShadow: t.isIOS
            ? [
                BoxShadow(
                  color: t.accent.withValues(alpha: 0.3),
                  blurRadius: 24,
                  offset: const Offset(0, 8),
                ),
              ]
            : null,
      ),
      child: Center(
        child: CustomPaint(
          size: Size.square(size * 0.55),
          // Mark color rides on the gradient (white on iOS, deep blue on MD).
          painter: _BrandGlyphPainter(mark: t.onAccent, dot: t.accent),
        ),
      ),
    );
  }
}

/// Paints the 4-square grid + accent dot from a 24x24 design viewBox.
class _BrandGlyphPainter extends CustomPainter {
  const _BrandGlyphPainter({required this.mark, required this.dot});

  final Color mark;
  final Color dot;

  @override
  void paint(Canvas canvas, Size size) {
    final s = size.width / 24.0; // viewBox 24 -> px scale
    final r = Radius.circular(1.5 * s);
    final paint = Paint()..isAntiAlias = true;

    void square(double x, double y, double opacity) {
      paint.color = mark.withValues(alpha: opacity);
      canvas.drawRRect(
        RRect.fromRectAndRadius(
          Rect.fromLTWH(x * s, y * s, 8 * s, 8 * s),
          r,
        ),
        paint,
      );
    }

    square(3, 3, 1.0);
    square(13, 3, 0.55);
    square(3, 13, 0.55);
    square(13, 13, 0.9);

    paint.color = dot;
    canvas.drawCircle(Offset(17 * s, 17 * s), 1.5 * s, paint);
  }

  @override
  bool shouldRepaint(_BrandGlyphPainter old) =>
      old.mark != mark || old.dot != dot;
}

File: apps/ios/NanoLink/Screens/ServerWelcomeScreen.swift
import SwiftUI

/// First-run screen: brand intro, product capabilities and the add-server CTA.
struct ServerWelcomeScreen: View {
    @Environment(\.nano) private var t
    @State private var showAddServer = false

    private let features: [(String, String, String)] = [
        ("chart.xyaxis.line", "welcome.featureMetricsTitle", "welcome.featureMetricsDesc"),
        ("terminal", "welcome.featureTerminalTitle", "welcome.featureTerminalDesc"),
        ("bell.badge", "welcome.featureAuditTitle", "welcome.featureAuditDesc"),
    ]

    var body: some View {
        Group {
            if t.desktop {
                if showAddServer {
                    AddServerScreen(onClose: { showAddServer = false })
                        .transition(.opacity)
                } else {
                    desktopContent
                        .transition(.opacity)
                }
            } else {
                touchContent
                    .sheet(isPresented: $showAddServer) {
                        NavigationStack { AddServerScreen() }
                    }
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(t.bg.ignoresSafeArea())
        .animation(.easeInOut(duration: 0.18), value: showAddServer)
    }

    private var desktopContent: some View {
        HStack(alignment: .center, spacing: 64) {
            VStack(alignment: .leading, spacing: 0) {
                BrandMark(size: 88)
                Text("NanoOps")
                    .font(.system(size: 38, weight: .bold))
                    .tracking(-1.1)
                    .foregroundColor(t.fg)
                    .padding(.top, 24)
                Text(tr("welcome.tagline"))
                    .font(.system(size: 16))
                    .foregroundColor(t.fg3)
                    .lineSpacing(6)
                    .fixedSize(horizontal: false, vertical: true)
                    .padding(.top, 10)

                NanoButton(tr("welcome.addServer"), icon: "plus", fullWidth: true) {
                    showAddServer = true
                }
                .frame(width: 220)
                .padding(.top, 32)

                Text(tr("welcome.chooseMethodHint"))
                    .font(.system(size: 12.5))
                    .foregroundColor(t.fg4)
                    .lineSpacing(4)
                    .frame(maxWidth: 330, alignment: .leading)
                    .padding(.top, 12)
            }
            .frame(width: 340, alignment: .leading)

            VStack(spacing: 14) {
                ForEach(Array(features.enumerated()), id: \.offset) { _, feature in
                    featureCard(feature, desktop: true)
                }
            }
            .frame(width: 460)
        }
        .padding(56)
        .frame(minWidth: 900, minHeight: 600)
        .frame(maxWidth: 1_020, maxHeight: .infinity, alignment: .center)
        .frame(maxWidth: .infinity)
    }

    private var touchContent: some View {
        VStack(spacing: 0) {
            ScrollView {
                VStack(alignment: t.isIOS ? .center : .leading, spacing: 0) {
                    BrandMark(size: t.isIOS ? 72 : 64)
                    Text("NanoOps")
                        .font(.system(size: t.isIOS ? 32 : 36, weight: t.displayWeight))
                        .tracking(t.displayTracking)
                        .foregroundColor(t.fg)
                        .padding(.top, 26)
                    Text(tr("welcome.tagline"))
                        .font(.system(size: 15.5))
                        .foregroundColor(t.fg3)
                        .multilineTextAlignment(t.isIOS ? .center : .leading)
                        .lineSpacing(5)
                        .padding(.top, 10)
                        .padding(.bottom, 28)

                    ForEach(Array(features.enumerated()), id: \.offset) { _, feature in
                        featureCard(feature, desktop: false)
                            .padding(.bottom, 10)
                    }
                }
                .frame(maxWidth: 460)
                .padding(EdgeInsets(top: 48, leading: 24, bottom: 8, trailing: 24))
                .frame(maxWidth: .infinity)
            }

            VStack(spacing: 0) {
                NanoButton(tr("welcome.addServer"), icon: "plus", fullWidth: true) {
                    showAddServer = true
                }
                if t.isIOS {
                    Text(tr("welcome.chooseMethodHint"))
                        .font(.system(size: 12))
                        .foregroundColor(t.fg4)
                        .multilineTextAlignment(.center)
                        .padding(.top, 14)
                }
            }
            .frame(maxWidth: 460)
            .padding(EdgeInsets(top: 8, leading: 24, bottom: 20, trailing: 24))
        }
    }

    private func featureCard(_ feature: (String, String, String), desktop: Bool) -> some View {
        NanoCard(
            padding: EdgeInsets(top: desktop ? 18 : 14, leading: desktop ? 18 : 14,
                                bottom: desktop ? 18 : 14, trailing: desktop ? 18 : 14),
            outlined: true
        ) {
            HStack(spacing: desktop ? 16 : 14) {
                NanoIconBox(icon: feature.0, size: desktop ? 48 : 40,
                            iconSize: desktop ? 22 : 20, fg: t.accent)
                VStack(alignment: .leading, spacing: desktop ? 4 : 2) {
                    Text(tr(feature.1))
                        .font(.system(size: desktop ? 16 : 15, weight: .semibold))
                        .foregroundColor(t.fg)
                    Text(tr(feature.2))
                        .font(.system(size: desktop ? 13.5 : 13))
                        .foregroundColor(t.fg3)
                }
            }
        }
    }
}

/// NanoOps gradient grid mark used by onboarding and settings.
struct BrandMark: View {
    let size: CGFloat
    @Environment(\.nano) private var t

    var body: some View {
        Image("BrandLogo")
            .resizable()
            .interpolation(.high)
            .scaledToFill()
            .frame(width: size, height: size)
            .clipShape(RoundedRectangle(cornerRadius: size * (t.isIOS ? 0.22 : 0.18), style: .continuous))
            .shadow(color: t.accent.opacity(t.desktop ? 0.16 : 0.26), radius: t.desktop ? 18 : 12, y: 8)
    }
}
